Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -252,55 +252,70 @@ public final class ManifestLoader: ManifestLoaderProtocol {
                     delegateQueue: queue,
                     fileSystem: fileSystem,
                     observabilityScope: observabilityScope
-                )
-
-                // Convert legacy system packages to the current target‐based model.
-                var products = parsedManifest.products
-                var targets = parsedManifest.targets
-                if products.isEmpty, targets.isEmpty,
-                    fileSystem.isFile(path.parentDirectory.appending(component: moduleMapFilename)) {
-                        products.append(ProductDescription(
-                        name: parsedManifest.name,
-                        type: .library(.automatic),
-                        targets: [parsedManifest.name])
-                    )
-                    targets.append(try TargetDescription(
-                        name: parsedManifest.name,
-                        path: "",
-                        type: .system,
-                        pkgConfig: parsedManifest.pkgConfig,
-                        providers: parsedManifest.providers
-                    ))
-                }
-
-                let manifest = Manifest(
-                    displayName: parsedManifest.name,
-                    path: path,
-                    packageKind: packageKind,
-                    packageLocation: packageLocation,
-                    defaultLocalization: parsedManifest.defaultLocalization,
-                    platforms: parsedManifest.platforms,
-                    version: version,
-                    revision: revision,
-                    toolsVersion: toolsVersion,
-                    pkgConfig: parsedManifest.pkgConfig,
-                    providers: parsedManifest.providers,
-                    cLanguageStandard: parsedManifest.cLanguageStandard,
-                    cxxLanguageStandard: parsedManifest.cxxLanguageStandard,
-                    swiftLanguageVersions: parsedManifest.swiftLanguageVersions,
-                    dependencies: parsedManifest.dependencies,
-                    products: products,
-                    targets: targets
-                )
-
-                try self.validate(manifest, toolsVersion: toolsVersion, observabilityScope: observabilityScope)
-
-                if observabilityScope.errorsReported {
-                    throw Diagnostics.fatalError
-                }
-
-                queue.async {
-                    completion(.success(manifest))
+                ) { parseResult in
+                    let parsedManifest : ManifestJSONParser.Result
+                    switch parseResult {
+                    case .success(let result):
+                        parsedManifest = result
+                    case .failure(let error):
+                        return queue.async {
+                            completion(.failure(error))
+                        }
+                    }
```

Comment:
this switch can probably be simplified to `let parsedManifest = try parseResult .get()` inside the do block